python/lsst/obs/base/ingest.py

```diff
@@ -274,6 +275,34 @@ class RawIngestTask(Task):
 
     _DefaultName: ClassVar[str] = "ingest"
 
+    obsInfoRequiredSubset: ClassVar[Dict[str, bool]] = {
+        "altaz_begin": False,
+        "boresight_rotation_coord": False,
+        "boresight_rotation_angle": False,
+        "dark_time": False,
+        "datetime_begin": True,
+        "datetime_end": True,
+        "detector_num": True,
+        "exposure_group": False,
+        "exposure_id": True,
+        "exposure_time": True,
+        "instrument": True,
+        "tracking_radec": False,
+        "object": False,
+        "observation_counter": False,
+        "observation_id": True,
+        "observation_reason": False,
+        "observation_type": True,
+        "observing_day": False,
+        "physical_filter": True,
+        "science_program": False,
+        "visit_id": False,
+    }
+    """Fields in the `ObservationInfo` that will be used by
+    ``makeExposureRecord`` in constructing an exposure record, and whether they
+    are required.
+    """
+
```

Comment:
I'd like to either make this a `classmethod` that returns a copy of the dictionary, a `classmethod` that returns a pair of required/optional sets, or a similar pair of `frozenset` class variables.  Having a multiple type as a class variable might tempt a derived class to modify it directly (changing the base class) instead of modifying its own copy.

Looking how it's used, I actually think one of the pair-of-sets variants might be a bit better than the dictionary one.